CRM/Core/Form.php

```diff
@@ -1051,7 +1058,12 @@ public function addDateRange($name, $from = '_from', $to = '_to', $label = 'From
    */
   public function addSelect($name, $props = array(), $required = FALSE) {
     if (!isset($props['entity'])) {
-      $props['entity'] = CRM_Utils_Api::getEntityName($this);
+      if(isset($this->entityName)){
```

Comment:
The style checker will probably flag this. Need to add a space between `if` and opening parenthesis. And another space before the `{`